qiskit_experiments/library/characterization/t1.py

```diff
@@ -98,7 +98,7 @@ def __init__(
         # Set experiment options
         self.set_experiment_options(delays=delays)
 
-    def circuits(self) -> List[QuantumCircuit]:
+    def circuits(self, qnum=0) -> List[QuantumCircuit]:
```

Comment:
I know this is much faster but `circuits()` is the base class method and signature shouldn't change. Indeed this breaks polymorphism of experiment class, e.g. user may think T2 also has this argument. Also `experiment.run` doesn't consider the `qnum` argument.

You can consider another approach of creating a pass manager that only performs virtual -> physical index mapping. Which must be lightweight compared with the default level0 pass manager (to reduce overhead you can directly run `layout`, i.e. `layout.run(circuits)`)

https://github.com/Qiskit-Extensions/qiskit-experiments/blob/fe4ccff973dae85fdff3416b253880051b71e6c3/qiskit_experiments/calibration_management/base_calibration_experiment.py#L317-L332

Unfortunately this code is not as performant as the proposed code change because of the initialization overhead of passes and pass manager instance. However this doesn't cause API break.

Reply:
I don't think there's anything forbidden by this idea in inheritance because the circuit call still behaves the same. You can absolutely add parameters to an inherited function. But this is also backwards compatibl. I also don't know why `run` needs to know anything about this. It's merely to allow the circuit to be created on any qubit if desired. I don't see why it's relevant whether a user will think T2 has this argument. Of course since this works really well I would argue it should be added to the T2 class as well.